tests/common/example_data_generator/base.py

```diff
@@ -0,0 +1,34 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable
+
+
+class ExampleDataGenerator(ABC):
+    @abstractmethod
+    def generate(self) -> Iterable[Dict[Any, Any]]:
+        pass
```

Comment:
nit; this comes down to taste as PEP-8 doesn't really take a stance on this, but my general impression (and the philosophy I subscribe to) is that ellipsis (`...`) should be used to indicate "code should go here", whereas `pass` should be put in places where "nothing should happen here". Therefore I recommend using ellipsis here instead of pass.
```suggestion
        ...
```